pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -262,12 +262,41 @@ func (service *AlertRuleService) CreateAlertRule(ctx context.Context, user ident
 	return rule, nil
 }
 
+// FilterOptions provides filtering for alert rule queries.
+// All fields are optional and will be applied as filters if provided.
+type FilterOptions struct {
+	ImportedPrometheusRule *bool
```

Comment:
Why is this a pointer? Couldn't we let it alone as a `bool` and let it default to false?

Reply:
Currently if it's `false` then it will return alert rules that are not imported from a Prometheus datasource, and if it is `nil` then no filtering is applied. We don't really need this, just thought it'd be nicer with this interface